compiler/rustc_hir/src/hir.rs

```diff
@@ -1517,6 +1517,26 @@ impl fmt::Debug for DotDotPos {
     }
 }
 
+#[derive(Debug, Clone, Copy, HashStable_Generic)]
+pub struct PatLit<'hir> {
```

Comment:
I somewhat dislike the name "Lit" here. It's more like "the subset of expressions we allow and treat similarly in patterns", since paths and const blocks aren't lits in the expr sense.